/**:
  ros__parameters:

    baud_rate: 921600                         # baud rate
    device_name: /dev/fximu                   # device serial name
    flow_control: none                        # serial flow control
    parity: none                              # serial parity
    stop_bits: "1"                            # serial stop bits

                                              # uint8 section
    gyroODR: 6                                # gyro measurement speed
    gyroFSR: 0                                # gyro sensitivity
    accelODR: 6                               # accelerometer measurement speed
    accelFSR: 0                               # acceletometer sensitivity

    notchFilterBW: 2                          # sensor level gyro notch filter bandwidth select
    notchFilterDIR: 7                         # sensor level gyro notch filter direction
    antiAliasFilterBW: 21                     # sensor level anti aliasing filter bandwidth

    outputDiv: 4                              # after N measurements, send data

    gyroUIFilterOrder: 2                      # gyro UI filter order
    gyroUIFilterIndex: 1                      # gyro UI filter index
    accelUIFilterOrder: 2                     # accel UI filter order
    accelUIFilterIndex: 1                     # accel UI filter index

    steadyLimit: 3                            # steady thresholds not exceeded

    magOdr: 0x4                               # 100hz
    magAvg: 0x2                               # 4
                                              # uint16 section
    timer0Sec: 4                              # timer0 seconds TODO: document
    timer1Sec: 5                              # initial calibration seconds
                                              # int16 section
    offsetGyroX: 0                            # gyro offset x
    offsetGyroY: 0                            # gyro offset y 
    offsetGyroZ: 0                            # gyro offset z
    offsetAccelX: 0                           # accel offset x
    offsetAccelY: 0                           # accel offset y
    offsetAccelZ: 0                           # accel offset z    
                                              # float section
    kDeltaAccelerationThreshold: 10.0         # steady detection threshold, acceleration change
    kDeltaAngularThreshold: 4.0               # steady detection threshold, angular change
    kAngularThreshold: 1.0                    # steady detection threshold, angular limit

    notchFilterFHZ: 1.449                     # notch filter fhz, consult datasheet

    filterGainAccel: 0.02                     # filter acceletometer gain
    filterGainMag: 0.001                      # filter magnetometer gain

    biasAlpha: 0.01                           # used for calculating biases
    gainAlpha: 0.95                           # used for calculating adaptive gain
    stableAlpha: 0.05                         # used for determining initial measurement
    magAlpha: 0.05                            # used for averaging magneto data

    enableNotchFilter: true                   # enable sensor level notch filter
    enableAAFilter: true                      # enable sensor level anti aliasing filter
    enableUIFilter: true                      # enable sensor level order and bandwidth filter

    enableSerial: false                       # enable device serial port for debug output
    enableInitialCalibration: false           # enable device initial calibration
    enableAdaptiveBias: true                  # enable device adaptive bias
    enableAdaptiveGain: true                  # enable device acceletometer adaptive gain

    enableMagnetometer: false                 # enable magnetometer device
    useMagnetometerData: false                # use magnetometer data for correcting orientation
    pubMagnetometerData: false                # publish magnetometer data

    imu_frame_id: imu_link                    # link name for imu message
    mag_frame_id: imu_link                    # link name for mag sensor message

    magBiasX: -5.50897506                     # magnetometer hard iron compensation
    magBiasY: -22.28865404
    magBiasZ: -12.588596

    magSoftA1: 0.99099148                     # magnetometer soft iron compensation
    magSoftA2: -0.00244458
    magSoftA3: 0.00608297
    magSoftB1: -0.00244458
    magSoftB2: 0.99692359
    magSoftB3: 0.00153484
    magSoftC1: 0.00608297 
    magSoftC2: 0.00153484 
    magSoftC3: 1.01225023

